### PR TITLE
Initial components

### PR DESCRIPTION
**Pull request by cameronviner**
#### What does this PR do? (please provide any background)

Creates a component that consists of a date picker and a time selection select option.

#### What tests does this PR have?

Unit tests

#### How can this be tested?

npm test to run the tests
npm start to see the examples

#### What gif best describes how you feel about this work?

---
#### Developer Definition of Done/Quality Checklist (for PR author to complete BEFORE code review):
- [x] I have checked the [contributing document](../blob/master/CONTRIBUTING.md) and I'm happy for this to be reviewed.

#### Software Engineer or Developer review:
- [x] I’ve witnessed the work behaving as expected (this could be on the authors machine or screencast).
- [x] I’ve checked for coding anti-patterns.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked all the tests are passing.

#### Software Engineer or project guru review:
- [ ] I’ve witnessed the work behaving as expected (this could be on the authors machine or screencast).
- [ ] I’ve checked for coding anti-patterns.
- [ ] I’ve checked for appropriate test coverage.
- [ ] I’ve checked all the tests are passing.

### CODE REVIEW

File: src/DateTimeGroup.jsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+    return this.props.dateExclusions.map(function(date) {
+      return moment(date);
+    });
+  },
+
+  render: function() {
+    var timePickerColumn = <span />;
+
+    // Should we mixin ReactIntl to ourselves so locales is pushed further down automatically?
```
Comment (hxpaul) [inline src/DateTimeGroup.jsx]: this looks like it should have a link to an issue

File: README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+Clone the repo and `npm install`.
+
+`npm start` will create and watchify an example which you can open in your browser, at `doc/example.html`
+
+`npm test` for the unit tests.
+
+`npm run lint` checks the code against our [guidelines](https://github.com/holidayextras/culture/blob/master/.eslintrc)
+
+`npm run coverage` gets coverage with istanbul, outputing to the `coverage` directory, and exiting nonzero if any metric is below 95%.
```
Comment (hxpaul) [inline README.md]: there's 100% coverage right now so you might as well up this, don't let anyone add any new non covered code

File: package.json [hunk 3]
```diff
@@ -0,0 +1,48 @@
+{
+  "name": "react-date-time-group",
+  "version": "1.0.0",
+  "description": "Datepicker with optional time",
+  "main": "dist/DateTimeGroup.js",
+  "scripts": {
+    "ci": "npm run lint && npm run coverage",
+    "coverage": "istanbul cover _mocha -- --compilers js:babel/register --require react-tests-globals-setup test/test* && istanbul check-coverage --statements 95 --branches 95 --functions 95 --lines 95",
```
Comment (hxpaul) [inline package.json]: this `--statements 95 --branches 95 --functions 95 --lines 95` can go into .istanbul.yml

File: package.json [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    "prepublish": "mkdir -p ./dist && babel ./src/DateTimeGroup.jsx > ./dist/DateTimeGroup.js",
+    "prestart": "ulimit -n 9999",
+    "start": "watchify -t babelify -t brfs doc/example.js -o doc/example-built -v",
+    "test": "mocha --require react-tests-globals-setup --compilers js:babel/register test/test*",
+    "lint": "make-up src test doc"
+  },
+  "browserify": {
+    "transform": ["brfs"]
+  },
+  "author": "Ollie Buck <ollie.buck@holidayextras.com>",
```
Comment (hxpaul) [inline package.json]: you can add yourself here too 
` "author": "Ollie Buck <ollie.buck@holidayextras.com>, Cameron Viner <etc>",`

File: doc/example.html [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    <meta charset="utf-8">
+    <title>DateTime group</title>
+    <link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/css/bootstrap.min.css" rel="stylesheet" integrity="sha256-MfvZlkHCEqatNoGiOXveE8FIwMzZg4W85qfrfIFBfYc= sha512-dTfge/zgoMYpP7QbHy4gWMEGsbsdZeCXz7irItjcC3sPUFtf0kuFbDz/ixG7ArTxmDjLXDmezHubeNikyKGVyQ==" crossorigin="anonymous">
+  </head>
+  <body>
+    <div id="container">
+    </div>
+  </body>
+
+  <script src="example-built"></script>
```
Comment (hxpaul) [inline doc/example.html]: I think this should be example-built.js minor point though

File: package.json [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  "name": "react-date-time-group",
+  "version": "1.0.0",
+  "description": "Datepicker with optional time",
+  "main": "dist/DateTimeGroup.js",
+  "scripts": {
+    "ci": "npm run lint && npm run coverage",
+    "coverage": "istanbul cover _mocha -- --compilers js:babel/register --require react-tests-globals-setup test/test* && istanbul check-coverage --statements 95 --branches 95 --functions 95 --lines 95",
+    "prepublish": "mkdir -p ./dist && babel ./src/DateTimeGroup.jsx > ./dist/DateTimeGroup.js",
+    "prestart": "ulimit -n 9999",
+    "start": "watchify -t babelify -t brfs doc/example.js -o doc/example-built -v",
```
Comment (hxpaul) [inline package.json]: I think this should be example-built.js minor point though

File: test/testDateTimeGroup.jsx [hunk 7]
```diff
@@ -0,0 +1,271 @@
...
+
+var expect = require('chai').expect;
+var sinon = require('sinon');
+
+var shallowRender = require('react-shallow-render');
+
+describe('DateTimeGroup', function() {
+  var clock, DateTimeGroup;
+
+  before(function() {
```
Comment (hxpaul) [inline test/testDateTimeGroup.jsx]: should this be beforeEach? Is there a difference?

Comment (shackpank) [inline test/testDateTimeGroup.jsx]: `before` only happens once, which maps more closely to the behaviour of `require`. If this used `beforeEach` we'd have to clear the module cache each time for things to work as expected.

Comment (hxpaul) [inline test/testDateTimeGroup.jsx]: thanks for clearing that up

**Comment by cameronviner** [comment]
Hey @hxpaul could you take another look when you get a second? I have addressed everything except the linking an issue comment as I need to catch up with Ollie

**Comment by hxpaul** [comment]
you've missed a comma after the author line in package.json - when that is fixed the ci will run I think, then i'm :+1: 
the issue link I mentioned would just be a comment in the code with a link to the github issue, if this is really something that needs to be revisited later. Happy to approve without that though

**Comment by GeorgeF72** [comment]
dev

**Comment by GeorgeF72** [comment]
Looks alright to me :+1:

**Comment by hxpaul** [comment]
:+1:

**Comment by hxpaul** [comment]
can't tick the checkboxes above for some reason

**Comment by cameronviner** [comment]
Awesome thanks for the reviews @hxpaul / @GeorgeF72 Ill merge this in now

**Pull request merged**